### PR TITLE
Fix test regressions due to C++ frontend  update

### PR DESCRIPTION
**Pull request by jketema**
See the individual commit messages.

Note that https://github.com/github/codeql/commit/324bd9e6f3656ed8f1a144e8c3060b59983e6f54 and https://github.com/github/codeql/commit/ef6c4c580a862c57d48e5085e5ddb9ce9a7d1765 need further investigation, but these are non-blocking.

### CODE REVIEW

File: cpp/ql/test/library-tests/syntax-zoo/builtin.c [hunk 1]
```diff
@@ -48,7 +48,7 @@ int builtin(int x, int y) {
 	acc++;
   }
 
-  acc += __atomic_fetch_min(&staticint, x+y, 0);
+  __atomic_fetch_min(&staticint, x+y, 0);
```
[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/test/library-tests/syntax-zoo/builtin.c:51]: Yup, seems like it. I just checked the EDG signature and didn't look at what Clang did. I'll add this to the list of issues I (potentially) need to report (this one definitely needs to be reported).

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Comment by jketema** [comment]
@MathiasVP No concerns about the issue @sashabu flagged up?

Note that I'm still waiting for the DCA experiment to complete (see the internal PR)

**Comment by MathiasVP** [comment]
> @MathiasVP No concerns about the issue @sashabu flagged up?

Oh, yeah. Sorry. I should have been more clear: LGTM except for the issue @sashabu highlighted.

**Comment by jketema** [comment]
@MathiasVP Thanks for clarifying. I'll check with Nick and Ian on the approach to take.

File: cpp/ql/test/library-tests/syntax-zoo/builtin.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/test/library-tests/syntax-zoo/builtin.c]: Ok there's something weird going on with version numbers in the frontend. I changed the clang version number in the test to match the one used in the previous version of the frontend. This fixes the issue without changing the test.

[the remark this answers is not in the document]
Comment (jketema) [reply cpp/ql/test/library-tests/syntax-zoo/builtin.c]: See: https://github.com/github/codeql/pull/8852/commits/73e4f0c044b82c4514e11f3114cb18e9098f3a52

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**